### DISCUSSION TITLE
Is there a way to stream music without the need for an external installation of Ffmpeg?

### DISCUSSION THREAD

**Discussion by Vincent-Cheong** [Rapptz/discord.py#6164] [General]
From the coding and deploying side, to install ffmpeg externally is not a desirable dependence. There is ffmpeg python module which can easily imported in code, can it be factored in or used together with discord.py to make a music bot more conveniently?

Or, are there any other methods for the bot to play or stream music without the need to install ffmpeg?

**Comment by imayhaveborkedit** [comment]
Any ffmpeg python package is just a set of cli bindings to the executable.  All they do is build console commands.  In short, no.  The reason ffmpeg is used is because it can decode just about any media you can throw at it to PCM (or opus) directly with very minimal effort.  There is no way anything has been written in pure python to do this, even before the obvious performance concerns.  At some level you're going to have to interact with native code, or offload it to another process in the case of lavalink.
